### PR TITLE
Use ref instead of state to save "should finalize" action

### PR DESCRIPTION
**Pull request by tnagorra**
- Error during entries save will cancel leads save
- Add cleanup logic on each termination branch

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

### CODE REVIEW

File: app/views/Project/EntryEdit/index.tsx [hunk 1]
```diff
@@ -308,49 +310,65 @@ function EntryEdit(props: Props) {
...
-                    authors: leadData?.authors?.map((author) => author.id),
-                });
+                shouldFinalizeRef.current = undefined;
             },
             onError: () => {
                 alert.show(
-                    'Failed to update source!',
+                    shouldFinalizeRef.current
+                        ? 'Failed to update source!'
+                        : 'Failed to update source!',
```
Comment (AdityaKhatri) [inline app/views/Project/EntryEdit/index.tsx]: why sir?

File: app/views/Project/EntryEdit/index.tsx [hunk 2]
```diff
@@ -635,32 +643,26 @@ function EntryEdit(props: Props) {
             formValidate,
             bulkUpdateEntries,
             projectId,
-            alert,
+            // alert,
```
Comment (AdityaKhatri) [inline app/views/Project/EntryEdit/index.tsx]: remove if not used

File: app/views/Project/EntryEdit/index.tsx [hunk 3]
```diff
@@ -635,32 +643,26 @@ function EntryEdit(props: Props) {
...
     );
 
     const handleSaveClick = useCallback(
         () => {
-            if (!entriesFormStale) {
-                handleLeadSave(false);
-            } else {
-                handleSubmit(false);
-            }
+            shouldFinalizeRef.current = true;
```
Comment (AdityaKhatri) [inline app/views/Project/EntryEdit/index.tsx]:
```suggestion
            shouldFinalizeRef.current = false;
```

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/Project/EntryEdit/index.tsx (hunk 1 above)
Comment (tnagorra) [reply app/views/Project/EntryEdit/index.tsx]: My bad!

**Pull request merged**